packages/astro/src/integration/vite-plugin-astro-config.ts

```diff
@@ -20,6 +20,13 @@ export function vitePluginAstroConfig(astroConfig: AstroConfig): VitePlugin {
         return resolvedVirtualModuleId;
       }
     },
+    config(config) {
+      // While Astro processes <script> tags by default, our control components
+      // which uses <script> tags and imports nanostores will not be processed by Astro.
+      // This ensures @clerk/astro/client is properly processed and bundled,
+      // resolving runtime import issues in these components.
+      config.optimizeDeps?.include?.push('@clerk/astro/client');
```

Comment:
More info on how Astro processed `<script>` tags [here](https://github.com/wobsoriano/astro-clerk-template).

The `@clerk/astro/client` export is imported by our control components, but Astro is not able to process it. Adding it to `optimizeDeps.include` will force astro to process it.